kele.el

```diff
@@ -937,6 +949,59 @@ Nil value for group denotes the core API."
   (let ((split (s-split "/" group-version)))
     (if (length= split 1) (list nil (car split)) split)))
 
+(cl-defun kele--list-resources (group version kind &key namespace context)
+  "Return the List of the given resource."
+  (when (and namespace
+             (not (kele--resource-namespaced-p
+                   kele--global-discovery-cache
+                   (kele--groupversion-string group version)
+                   kind)))
+    (signal 'user-error '()))
+
+  (-if-let* (((&alist 'port port) (kele--ensure-proxy
+                                   (or context (kele-current-context-name))))
+             (url (format "http://localhost:%s/%s/%s"
+                          port
+                          (if group
+                              (format "apis/%s/%s" group version)
+                            (format "api/%s" version))
+                          kind))
+             (data (kele--retry (lambda () (plz 'get url :as #'json-read))))
+             (filtered-items (->> (append  (alist-get 'items data) '())
+                                  (-filter (lambda (item)
+                                             (if (not namespace) t
+                                               (let-alist item
+                                                 (equal .metadata.namespace namespace))))))))
+      (progn
+        (setf (cdr (assoc 'items data)) filtered-items)
+        data)
+    (signal 'error (format "Failed to fetch %s/%s/%s" group version kind))))
+
+(cl-defun kele--fetch-resources (group version kind &key namespace context)
```

Comment:
`--list-resources`, `--fetch-resources`, and `--fetch-resource-names` largely overlap. Let's DRY things out a bit.